crates/cxx-qt-gen/src/generator/cpp/qobject.rs

```diff
@@ -113,19 +113,12 @@ impl GeneratedCppQObject {
             has_qobject_macro: qobject.has_qobject_macro,
         };
 
-        // Build the base class
-        let base_class = qobject.base_class.clone().unwrap_or_else(|| {
-            // If there is a QObject macro then assume the base class is QObject
-            if qobject.has_qobject_macro {
-                "QObject".to_string()
-            } else {
-                // CODECOV_EXCLUDE_START
-                unreachable!(
-                    "Cannot have an empty #[base] attribute  with no #[qobject] attribute"
-                );
-                // CODECOV_EXCLUDE_STOP
-            }
-        });
+        let base_class = if let Some(ident) = &qobject.base_class {
+            type_names.lookup(ident)?.cxx_qualified()
+        } else {
+            "QObject".to_string()
```

Comment:
this part is losing the `if qobject.has_qobject_macro` and then the unreachable block? That should still be in this else block i think

Reply:
Didn't think it was necessary anymore as we check for qobject and base better, but will add it back

Reply:
Think it's a sanity check that we don't end up in a situation where there was no #[qobject] macro on a type without a #[base]. 

There is the question over whether we should require #[qobject] on all types in RustQt, do we require that already i forget? As there is the question over supporting #[qgadget] later. But maybe for now we could require all types to have qobject macro and then optionally have a base class? Then that `has_qobject_macro` could go and so could this block? Unless there is another reason for it?

Reply:
I remember now, the `has_qobject_macro` was so that we can support https://doc.qt.io/qt-6/qquickframebufferobject-renderer.html which you need to inherit from but has now QObject as a base class